gsa/src/web/pages/login/__tests__/loginpage.js

```diff
@@ -0,0 +1,159 @@
+/* Copyright (C) 2019 Greenbone Networks GmbH
+ *
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+import React from 'react';
+
+import Logger from 'gmp/log';
+
+import {rendererWith, fireEvent, waitForElement} from 'web/utils/testing';
+
+import LoginPage from '../loginpage';
+
+Logger.setDefaultLevel('silent');
+
+describe('LoginPagetests', () => {
+  test('should render Loginpage', () => {
+    const isLoggedIn = jest.fn().mockReturnValue(false);
+    const clearToken = jest.fn();
+    const gmp = {isLoggedIn, clearToken, settings: {}};
+
+    const {render} = rendererWith({gmp, router: true, store: true});
+
+    const {baseElement} = render(<LoginPage />);
+
+    expect(baseElement).toMatchSnapshot();
+  });
+
+  test('should allow to login with username and password', () => {
+    const login = jest.fn().mockReturnValue(
```

Comment:
Could be simplified by using [mockResolvedValue](https://jestjs.io/docs/en/mock-function-api#mockfnmockresolvedvaluevalue) directly